src/step.rs

```diff
@@ -33,21 +32,21 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         self.memory.clear_packed();
         self.inc_step_counter_and_check_limit(1)?;
         if self.stack.is_empty() {
```

Comment:
Instead of triggering this on "stack empty" (and duplicating code), what about making the main cleanup StackPopCleanup::Tls(0)?